cli/src/config.rs

```diff
@@ -825,9 +825,8 @@ impl CommandNameAndArgs {
     pub fn split_name_and_args(&self) -> (Cow<'_, str>, Cow<'_, [String]>) {
         match self {
             Self::String(s) => {
-                // Handle things like `EDITOR=emacs -nw` (TODO: parse shell escapes)
-                let mut args = s.split(' ').map(|s| s.to_owned());
-                (args.next().unwrap().into(), args.collect())
+                let mut parts = shlex::Shlex::new(s);
```

Comment:
Maybe better to check `had_error` and fall back to original string (or space-separated args)? The command execution would be more likely to fail, and the user can notice the problem.
https://docs.rs/shlex/1.3.0/shlex/bytes/struct.Shlex.html#structfield.had_error